src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -106,12 +110,15 @@ public int Count
             }
         }
 
-        public Enumerator GetEnumerator()
+        public T this[int index]
         {
-            if (_array == null)
-                ThrowHelper.ThrowInvalidOperationException(ExceptionResource.InvalidOperation_NullArray);
-            Contract.EndContractBlock();
+            get { return _array[_offset + index]; }
```

Comment:
@jkotas OK, not disagreeing. If someone finds the extra overhead to be a problem maybe they can use `Unsafe.Add`.